Sources/BraveWallet/Crypto/Stores/SendTokenStore.swift

```diff
@@ -470,21 +475,18 @@ public class SendTokenStore: ObservableObject {
       return
     }
     let amount = (token.isErc721 || token.isNft) ? "1" : amount
-    walletService.selectedCoin { [weak self] coin in
-      guard let self = self else { return }
-      self.keyringService.selectedAccount(coin) { selectedAccount in
-        guard let selectedAccount = selectedAccount else {
-          completion(false, "An Internal Error")
-          return
-        }
-        switch coin {
-        case .eth:
-          self.sendTokenOnEth(amount: amount, token: token, fromAddress: selectedAccount, completion: completion)
-        case .sol:
-          self.sendTokenOnSol(amount: amount, token: token, fromAddress: selectedAccount, completion: completion)
-        default:
-          break
-        }
+    keyringService.allAccounts { allAccounts in
+      guard let selectedAccount = allAccounts.selectedAccount else {
+        completion(false, "An Internal Error")
+        return
+      }
+      switch selectedAccount.coin {
+      case .eth:
+        self.sendTokenOnEth(amount: amount, token: token, fromAddress: selectedAccount.address, completion: completion)
+      case .sol:
+        self.sendTokenOnSol(amount: amount, token: token, fromAddress: selectedAccount.address, completion: completion)
+      default:
+        completion(false, "An Internal Error")
```

Comment:
just realized we didnt localize this error message lol

Reply:
This error message [doesn't actually get used](https://github.com/brave/brave-ios/blob/development/Sources/BraveWallet/Crypto/BuySendSwap/SendTokenView.swift#L237), we use a generic `"We currently cannot proceed with your transaction"` message.